### DISCUSSION TITLE
Unexpected number of bytes (chunk size) in encoding?

### DISCUSSION THREAD

**Discussion by thorwhalen** [otosense/recode#7] [Q&A] [answered]
Noticed this unexpected behavior:

```python
import recode
dh_codec = recode.mk_codec('dh')
assert len(dh_codec.encode([[1.0, 1]])) == 10

hd_codec = recode.mk_codec('hd')
assert len(hd_codec.encode([[1, 1.0]])) == 16
```

Which can be traced down to the builtin `struct` itself:

```python
import struct
assert struct.calcsize('h') == 2
assert struct.calcsize('d') == 8
assert struct.calcsize('dh') == 10 == 2 + 8  # expected!

assert struct.calcsize('hd') == 16  # didn't expect this!
```

What's happening here?

How can I get what I expect? (10 in both cases.)

**Comment by charlie-dufort** [comment, marked as answer]
@sylvainbonnot explained it had to do with this optional prefix character (`'@'`, `'='`, `'<'`, `'>'`, `'!'`) which controls the byte order, size and alignment. 
If the first character is not one of these, `'@'` is assumed.
Using `'='` removes the "unexpected behavior":
```
>>> struct.calcsize('=hd') == struct.calcsize('=dh') == 10
True
```
